tangelo/linq/translator/translate_qiskit.py

```diff
@@ -103,6 +115,48 @@ def translate_qiskit(source_circuit):
             (GATE_QISKIT[gate.name])(target_circuit, gate.target[0], gate.target[0])
         else:
             raise ValueError(f"Gate '{gate.name}' not supported on backend qiskit")
+
+    return target_circuit
+
+
+def translate_c_from_qiskit(source_circuit) -> Circuit:
+    """Take in a qiskit.QuantumCircuit, return an equivalent Circuit
+
+    Args:
+        source_circuit (qiskit.QuantumCircuit): quantum circuit in the abstract qiskit.QuantumCircuit format.
+
+    Returns:
+        Circuit: the corresponding quantum Circuit.
+    """
+    import qiskit
+
+    GATE_QISKIT = get_qiskit_gates()
+    qi = source_circuit._qubit_indices
+    inv_GATE_QISKIT = {v.__name__: k for k, v in GATE_QISKIT.items()}
+
+    gates = []
+    for gate in source_circuit:
+        name = inv_GATE_QISKIT[gate.operation.name]
+        if name in {"H", "X", "Y", "Z", "S", "T"}:
+            gates += [Gate(name, qi[gate.qubits[0]].index)]
+        elif name in {"RX", "RY", "RZ", "PHASE"}:
+            gates += [Gate(name, qi[gate.qubits[0]].index, parameter=gate.operation.params[0])]
+        elif name in {"CRX", "CRY", "CRZ", "CPHASE"}:
+            gates += [Gate(name, qi[gate.qubits[1]].index, control=qi[gate.qubits[0]].index, parameter=gate.operation.params[0])]
+        elif name in {"CNOT", "CH", "CX", "CY", "CZ"}:
+            gates += [Gate(name, qi[gate.qubits[1]].index, control=qi[gate.qubits[0]].index)]
+        elif name in {"SWAP"}:
+            gates += [Gate(name, [qi[gate.qubits[0]].index, qi[gate.qubits[1]].index])]
+        elif name in {"CSWAP"}:
+            gates += [Gate(name, [qi[gate.qubits[1]].index, qi[gate.qubits[2]].index], control=qi[gate.qubits[0]].index)]
+        elif name in {"XX"}:
+            gates += [Gate(name, [qi[gate.qubits[0]].index, qi[gate.qubits[1]].index], parameter=gate.operation.params[0])]
+        elif name in {"MEASURE"}:
+            gates += [Gate(name, qi[gate.qubits[0]].index)]
+        else:
+            raise ValueError(f"Gate '{gate.name}' not supported on backend qiskit")
```

Comment:
Shouldn't this say that the gate.name is not support in Tangelo instead ? Since gate is froma  source circuit in qiskit format, qiskit should obviously support it.

Also, any name is supported in our Tangelo Circuit. We don't check for names since it is an"abstract" format (POTATO gate example etc). I'm not sure this check is warranted at all